shared/src/model/exercise/question.js

```diff
@@ -114,22 +114,32 @@ class ExerciseQuestion extends BaseModel {
   }
 
   @action setExclusiveFormat(name) {
+      //reset formats and answers
+      this.formats = [];
+      if(this.answers.length > 0) {
+          this.answers = [];
+      }
       if (name == 'open-ended') { name = 'free-response'; }
-
-      let formats = without(map(this.formats, 'value'), ...keys(ExerciseQuestion.FORMAT_TYPES));
-      formats.push(name);
-      if ('true-false' === name) {
-          formats = without(formats, 'free-response');
-          if (this.answers.length == 0) {
-              this.answers.push({ content_html: 'True' });
-              this.answers.push({ content_html: 'False' });
-          }
-      } else if ('multiple-choice' === name && !formats.includes('free-response')) {
-          formats = formats.concat('free-response');
-      } else if (name == 'free-response') {
-          this.exercise.onQuestionFreeResponseSelected(this);
+      this.formats.push(name);
+
+      switch(name) {
+          case 'free-response':
+              this.exercise.onQuestionFreeResponseSelected(this);
+              break;
+          case 'multiple-choice':
+              if(!this.formats.includes('free-response')) {
+                  this.formats.push('free-response')
+              }
+              break;
+          case 'true-false':
+              if (this.answers.length == 0) {
+                  this.answers.push({ content_html: 'True' });
+                  this.answers.push({ content_html: 'False' });
+              }  
+              break;
       }
-      this.formats = uniq(formats).sort();
+
+      uniq(this.formats).sort();
```

Comment:
we do not want modify the answers until the exercise is saved.  Image the case of an editor who types up very long answers, then inadvertently hits the 'free-response' radio input, quickly switches back to multiple-choice and has lost all their work.